src/sprite_weapon.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EP_SPRITE_WEAPON_H
+#define EP_SPRITE_WEAPON_H
+
+// Headers
+#include "sprite_battler.h"
+#include "async_handler.h"
+
+class Game_Battler;
+class Game_Actor;
+
+/**
+ * Sprite_Weapon class, used for weapon battle sprites
+ */
+class Sprite_Weapon : public Sprite_Battler {
```

Comment:
Why does this inherit from `Sprite_Battler` and not `Sprite`?

Reply:
Well, when I started implementing the CBA weapons I took `Sprite_Actor` as a base which inherits from `Sprite_Battler`.

Reply:
Considering that you do not use ``ResetZ`` anymore this is a legitimate request.
Just change it to ``Sprite``. It should work. Can be squashed into commit "Remove ResetZ function from sprite_weapon "

Reply:
Did the requested changed now.